### ISSUE TITLE
Potential bugs or improper implementation of dynamic NTK.

### ISSUE THREAD

**Issue by Dinghow** [QwenLM/Qwen#401]
Thanks for your introduction of dynamic NTK and logN attention to make Qwen support long sequence inference. I have several questions about the implementation of dynamic NTK.
1. Qwen-7B caches the ntk value in attention while 14B caches it in RoPE,  both way may face two problems in inference.
    - For static batching inference, if there is an outlier long sequence in the batch (e.g. bs=8, 7 prompts with length 0.5k, while an outlier one 4k), the ntk value for this batch will all use the longest one with padding, which may cause error on RoPE.
    - For continuous batching inference, it's hard to follow this ntk cache type since the batch is dynamic. In VLLM, they directly set this ntk value as a param provided by the user in the inference request. (A better way to cover this case is to maintain an ntk value for each request)
2. The NTK value formula of Qwen is $2^{\lfloor{log_2(\frac{full\ seq\ len}{long\ seq\ len\ threshold})+1}\rfloor}-1$, I wonder what if the decode length exceeds the floor of $log_2$ (e.g. the prompt length is 2k, while decoding to a position more than 4k, the ntk rope for 4k+ pos will have error).

**Comment by logicwong** [comment]
关于static batching inference，现在的实现是为batch里的每条样本按实际长度计算ntk，所以不会有啥问题。其它问题我们再一块看看吧
